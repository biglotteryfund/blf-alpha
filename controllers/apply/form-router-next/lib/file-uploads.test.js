/* eslint-env jest */
'use strict';
const { prepareFilesForUpload } = require('./file-uploads');

test('determine files to upload', () => {
    const result = prepareFilesForUpload(
        [
            { name: 'bankStatement', type: 'file' },
            { name: 'anotherField', type: 'text' }
        ],
        {
            bankStatement: {
                size: 13264,
                name: 'example.pdf',
                type: 'application/pdf'
            }
        }
    );

    expect(result).toMatchSnapshot();
});

test('trim file names for upload', () => {
    const result = prepareFilesForUpload(
        [
            { name: 'bankStatement', type: 'file' },
            { name: 'anotherField', type: 'text' }
        ],
        {
            bankStatement: {
                size: 13264,
                name: '   example.pdf  ',
                type: 'application/pdf'
            }
        }
    );

    expect(result).toMatchSnapshot();
});
