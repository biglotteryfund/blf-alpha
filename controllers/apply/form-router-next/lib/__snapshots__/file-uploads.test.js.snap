// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`determine files to upload 1`] = `
Object {
  "filesToUpload": Array [
    Object {
      "fieldName": "bankStatement",
      "fileData": Object {
        "name": "example.pdf",
        "size": 13264,
        "type": "application/pdf",
      },
    },
  ],
  "valuesByField": Object {
    "bankStatement": Object {
      "filename": "example.pdf",
      "size": 13264,
      "type": "application/pdf",
    },
  },
}
`;

exports[`trim file names for upload 1`] = `
Object {
  "filesToUpload": Array [
    Object {
      "fieldName": "bankStatement",
      "fileData": Object {
        "name": "example.pdf",
        "size": 13264,
        "type": "application/pdf",
      },
    },
  ],
  "valuesByField": Object {
    "bankStatement": Object {
      "filename": "example.pdf",
      "size": 13264,
      "type": "application/pdf",
    },
  },
}
`;
